src/lib/server/events/gcal.ts

```diff
@@ -305,113 +307,23 @@ export function zonedDateTimeFromICALDateTime(
-    return null;
-  }
-
-  const title =
-    icalEvent['SUMMARY'] !== undefined ? icalEvent['SUMMARY'].replace(/\\/g, '') : 'Unnamed Event';
-
-  const recurring = parseRRULE(icalEvent['RRULE']);
-
-  const dtStart = zonedDateTimeFromICALDateTime(icalEvent['DTSTART'], referenceDate.timeZone);
-  const dtEnd = zonedDateTimeFromICALDateTime(icalEvent['DTEND'], referenceDate.timeZone);
-  const date = dtStart.toString();
-  const month = dtStart.toLocaleString('en-US', { month: 'long' });
-  const day = dtStart.day;
-
-  const startTime = dtStart.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric' });
-  const endTime = dtEnd.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric' });
-
-  const hasStarted = Temporal.ZonedDateTime.compare(referenceDate, dtStart) >= 0;
-  const hasEnded = Temporal.ZonedDateTime.compare(referenceDate, dtEnd) >= 0;
-  const duration = dtEnd.since(dtStart).minutes + ' minutes';
-
-  const { description, variables } = parseDescription(icalEvent['DESCRIPTION']);
-
-  const { location, meetingLink } = parseLocation(
-    icalEvent['LOCATION'],
-    variables.get('ACM_LOCATION')
-  );
-
-  const isPinned = parseBool(variables.get('ACM_PINNED'));
-
-  const slug = makeEventSlug(title, dtStart);
-
-  const selfLink = makeEventLink(slug);
-
-  const summary = produceSummary(title, description, selfLink);
-
-  /**
-   * Team ID hosting the event is stored in the description. If it's not there,
-   * it's a "general" event.
-   *
-   * We check the variable name `ACM_TEAM` first, then fall back to `ACM_PATH` which is deprecated.
-   * We expect new events to use `ACM_TEAM` and ancient events to use `ACM_PATH`.
-   */
-  const teamId = (variables.get('ACM_TEAM') ?? variables.get('ACM_PATH'))?.toLowerCase().trim();
-
-  let team = TEAMS.general;
-  switch (teamId) {
-    case 'ai': {
-      team = TEAMS.ai;
-      break;
-    }
-    case 'algo': {
-      team = TEAMS.algo;
-      break;
-    }
-    case 'design': {
-      team = TEAMS.design;
-      break;
-    }
-    case 'dev': {
-      team = TEAMS.dev;
-      break;
-    }
-    case 'special-events': {
-      team = TEAMS['special-events'];
-      break;
-    }
-  }
-
-  const thirdPartyCalendarLocation = location === 'Discord' ? selfLink : location;
-  const thirdPartyCalendarArgs = [
-    title,
-    summary,
-    thirdPartyCalendarLocation,
-    dtStart,
-    dtEnd,
-  ] as const;
-
-  const calendarLinks = {
-    google: makeGoogleCalendarLink(...thirdPartyCalendarArgs).toString(),
-    outlook: makeOutlookCalendarLink(...thirdPartyCalendarArgs).toString(),
-  };
-
-  return {
-    month,
-    day,
-    startTime,
-    endTime,
-    date,
-    hasStarted,
-    hasEnded,
-    isPinned,
-    duration,
-    location,
-    title,
-    summary,
-    description,
-    meetingLink,
-    slug,
-    selfLink,
-    recurring,
-    team,
-    calendarLinks,
-  };
+export async function listUpcomingEvents() {
```

Comment:
I think this function should return `GCalEvent[]`.
```suggestion
export async function listUpcomingEvents(): Promise<GCalEvent[]> {
```

Reply:
Talked to Ethan, don't need this type annotation but updated return value of this and the type `GCalEvent` to be easier to use and read